hw2/01-coins.js

```diff
@@ -1,7 +1,44 @@
 /** Exercise 01 - Coins **/
 
+//Brian Koehler
+//Full Stack - Portland State
+
+const pluralize = (num, plural, single) => {
+  return num > 1 ? plural : single;
+};
+
 const calculateChange = (input) => {
   // Add your code here
+  if (input === 0 || input > 10) return "No Change Available";
+  if (isNaN(input)) return "Input Numbers Only";
+
+  let curChange = input;
+  let curDollar = Math.floor(curChange / 1);
+  curChange -= curDollar * 1;
+  let curQtr = Math.floor(curChange / 0.25);
+  curChange -= curQtr * 0.25;
+  let curDime = Math.floor(curChange / 0.1);
+  curChange -= curDime * 0.1;
+  let curNickel = Math.floor(curChange / 0.05);
+  curChange -= curNickel * 0.05;
+  let curPenny = Math.ceil(curChange / 0.01);
+  curChange -= curPenny * 0.01;
+
+  return (
+    "$" +
+    input +
+    " ==> " +
+    curDollar +
+    pluralize(curDollar, " dollars, ", " dollar, ") +
+    curQtr +
+    pluralize(curQtr, " quarters, ", " quarter, ") +
+    curDime +
+    pluralize(curDime, " dimes, ", " dime, ") +
+    curNickel +
+    pluralize(curNickel, " nickels, ", " nickel, ") +
+    curPenny +
+    pluralize(curPenny, " pennies.", " penny.")
```

Comment:
Hello,

The logic you used is really good. However, there is one issue that I found when I ran your code. For the below two sample cases the number of pennies it should output is slightly wrong. For example:
$4.62 ==> 4 dollars, 2 quarters, 1 dime, 0 nickel, 3 pennies.
$9.74 ==> 9 dollars, 2 quarters, 2 dimes, 0 nickel, 5 pennies.
In both the sample cases the number of pennies is greater than the actual by 1.